https://sourceforge.net/p/mediatomb/discussion/440751/thread/a7b5e7cf/?limit=25#7698

http://www.mail-archive.com/debian-bugs-dist%40lists.debian.org/msg1035589.html

Bug#677959: mediatomb: fails to build against current version of libavformat


--- a/src/metadata/ffmpeg_handler.cc.orig	2015-01-01 21:34:17.324147961 +0100
+++ b/src/metadata/ffmpeg_handler.cc	2015-01-01 21:39:34.607417634 +0100
@@ -107,8 +107,8 @@
 		return;
 	for (const mapping_t *m = mapping; m->avname != NULL; m++)
 	{
-		AVMetadataTag *tag = NULL;
-		tag = av_metadata_get(pFormatCtx->metadata, m->avname, NULL, 0);
+		AVDictionaryEntry *tag = NULL;
+		tag = av_dict_get(pFormatCtx->metadata, m->avname, NULL, 0);
 		if (tag && tag->value && tag->value[0])
 		{
 			log_debug("Added metadata %s: %s\n", m->avname, tag->value);
@@ -279,7 +279,7 @@
     int x = 0;
     int y = 0;
 
-	AVFormatContext *pFormatCtx;
+	AVFormatContext *pFormatCtx = avformat_alloc_context();
 	
 	// Suppress all log messages
 	av_log_set_callback(FfmpegNoOutputStub);
@@ -288,14 +288,14 @@
     av_register_all();
 
     // Open video file
-    if (av_open_input_file(&pFormatCtx, 
-                          item->getLocation().c_str(), NULL, 0, NULL) != 0)
+    if (avformat_open_input(&pFormatCtx,
+                          item->getLocation().c_str(), NULL, NULL) != 0)
         return; // Couldn't open file
 
     // Retrieve stream information
-    if (av_find_stream_info(pFormatCtx) < 0)
+    if (avformat_find_stream_info(pFormatCtx,NULL) < 0)
     {
-        av_close_input_file(pFormatCtx);
+        avformat_close_input(&pFormatCtx);
         return; // Couldn't find stream information
     }   
 	// Add metadata using ffmpeg library calls
@@ -304,7 +304,7 @@
 	addFfmpegResourceFields(item, pFormatCtx, &x, &y);
 	
     // Close the video file
-    av_close_input_file(pFormatCtx);
+    avformat_close_input(&pFormatCtx);
 }
 
 Ref<IOHandler> FfmpegHandler::serveContent(Ref<CdsItem> item, int resNum, off_t *data_size)
